pkg/network/protocols/http/stats.go

```diff
@@ -126,6 +126,9 @@ type RequestStat struct {
 
 	// Dynamic tags (if attached)
 	DynamicTags []string
+
+	// CanRelease is true when the struct is no longer referenced
+	CanRelease bool
```

Comment:
the current design frees the memory only if we called CombineWith, but there are other flows in which we don't call combineWith but still must release the resources